ext/snmp/snmp.c

```diff
@@ -676,8 +676,13 @@ static bool php_snmp_parse_oid(
 		objid_query->vars = (snmpobjarg *)safe_emalloc(sizeof(snmpobjarg), zend_hash_num_elements(oid_ht), 0);
 		objid_query->array_output = (st & SNMP_CMD_SET) == 0;
 		ZEND_HASH_FOREACH_VAL(oid_ht, tmp_oid) {
-			convert_to_string(tmp_oid);
-			objid_query->vars[objid_query->count].oid = Z_STRVAL_P(tmp_oid);
+			zend_string *tmp = zval_try_get_string(tmp_oid);
+			if (!tmp) {
+				efree(objid_query->vars);
+				return false;
+			}
+			objid_query->vars[objid_query->count].oid = ZSTR_VAL(tmp);
```

Comment:
You're pointing to the string contents here, but releasing the string the next line. This will cause a UAF for non-interned strings. You don't see that in your test because those are all interned.
Furthermore, there are other convert_to_ calls in this code.

ext-ldap had a similar situation. There we came up with the idea of only releasing the strings after the ldap request was done, and then use a pointer hack to release the strings afterwards: https://github.com/php/php-src/blob/7b9b832be5147e679f9098a6e3e423c5ee3d03b2/ext/ldap/ldap.c#L257-L260

Reply:
smart hack thanks I ll try to apply.